contributors/devel/development.md

```diff
@@ -168,69 +278,87 @@ development environment, please [set one up](http://golang.org/doc/code.html).
 | 1.17 - 1.18    | 1.13.9      |
 | 1.18+          | 1.14.4      |
 
-Ensure your GOPATH and PATH have been configured in accordance with the Go
-environment instructions.
-
-#### Upgrading Go
+##### A Note on Changing Go Versions
 
-Upgrading Go requires specific modification of some scripts and container
-images.
+If you have already compiled Kubernetes but are now trying with a
+different version of Go, you will need to make some modifications to
+your source tree. Update the Go version in the following locations:
 
-- The image for cross compiling in [build/build-image/cross].
-  The `VERSION` file and `Dockerfile`.
-- The cross tag `KUBE_BUILD_IMAGE_CROSS_TAG` in [build/common.sh].
+- The `VERSION` file for cross compiling in [build/build-image/cross].
 - The `go_version` in the [`go_register_toolchains`](https://git.k8s.io/kubernetes/build/root/WORKSPACE) bazel rule.
-- The desired Go version in
-  [test/images/Makefile](https://git.k8s.io/kubernetes/test/images/Makefile).
+- The desired Go version in [test/images/Makefile](https://git.k8s.io/kubernetes/test/images/Makefile).
 
-### PyYAML
+#### PyYAML
 
 Some Kubernetes verification tests use [PyYAML](https://pyyaml.org/) and it therefore needs to be installed to successfully run all verification tests in your local environment. 
-You can use the [PyYAML documentation](https://pyyaml.org/wiki/PyYAMLDocumentation) to find the installation instructions.
+You can use the
+[PyYAML documentation](https://pyyaml.org/wiki/PyYAMLDocumentation) to
+find the installation instructions for your platform.
 
-### Quick Start
+**Note:** If you are running macOS, you may need to use the `pip3`
+command instead of the `pip` command to install PyYAML.
 
-The following section is a quick start on how to build Kubernetes locally, for more detailed information you can see [kubernetes/build](https://git.k8s.io/kubernetes/build/README.md). Before you start you will need to clone a local Kubernetes code base, refer to [GitHub workflow](#github-workflow) for details on how to set this up.
+#### Cloning the Kubernetes Git Repository
 
-The best way to validate your current setup is to build a small part of Kubernetes. This way you can address issues without waiting for the full build to complete. To build a specific part of Kubernetes use the `WHAT` environment variable to let the build scripts know you want to build only a certain package/executable.
+You are now ready to clone the Kubernetes git repository. See the [GitHub Workflow](/contributors/guide/github-workflow.md) document from the Contributor Guide for instructions.
+
+#### etcd
+
+To test Kubernetes, you will need to install etcd, a consistent and highly-available key value store. As of Kubernetes 1.18, the required etcd version is at least 3.2.10. To install a local version of etcd, run the following command in your Kubernetes working directory.
```

Comment:
I think we can probably avoid mentioning the etcd version here because it might get stale (we've faced this issue with the go versions).

We could probably just ask the reader to install the appropriate version of etcd by using this script.